packages/jest-runtime/src/should_instrument.js

```diff
@@ -36,18 +36,23 @@ export default function shouldInstrument(
   }
 
   if (
-    config.testRegex &&
-    config.testRegex.some(regex => new RegExp(regex).test(filename))
+    !config.testPathIgnorePatterns ||
+    !config.testPathIgnorePatterns.some(pattern => filename.match(pattern))
   ) {
```

Comment:
there's the same check a bit further down this file: https://github.com/facebook/jest/blob/f38b0c9d17f92f3e0f144087db6264d585bcbeed/packages/jest-runtime/src/should_instrument.js#L79-L84

maybe we can just move it up to this place?

Reply:
@thymikee `testPathIgnorePatterns` vs `coveragePathIgnorePatterns`, no?

Reply:
@thymikee 
@SimenB 

Thank you for your review.

I think that it is determined by `testRegex`, `testMatch` and `testPathIgnorePatterns` whether file is test file or not.